ord_schema/message_helpers.py

```diff
@@ -98,6 +98,58 @@ def build_compound(smiles=None, name=None, amount=None, role=None,
     return compound
 
 
+def build_solution(solute, solvent, concentration):
+    """Helps define components for stock solution inputs with a single solute
+    and a single solvent compound.
+
+    Args:
+        solute: Compound with identifiers, roles, etc.; this argument is
+            modified in place to define an amount in moles.
+        solvent: Compound with identifiers, roles, etc. and defined volume.
+        concentration: string defining solute concentration.
+
+    Returns:
+        List of Compounds to assign to a repeated components field.
+    """
+    # Get solvent volume in liters.
+    if not solvent.HasField('volume') or not solvent.volume.value:
+        raise ValueError('solvent must have defined volume')
+    volume = solvent.volume.value
+    if solvent.volume.units == solvent.volume.LITER:
+        volume_liter = volume
+    elif solvent.volume.units == solvent.volume.MILLILITER:
+        volume_liter = volume * 1e-3
+    elif solvent.volume.units == solvent.volume.MICROLITER:
+        volume_liter = volume * 1e-6
+    else:
+        raise ValueError('solvent units not recognized by build_solution')
+    # Get solute concentration in molar.
+    resolver = units.UnitResolver(
+        unit_synonyms=units.CONCENTRATION_UNIT_SYNONYMS,
+        forbidden_units={},)
+    concentration_pb = resolver.resolve(concentration)
+    if concentration_pb.units == concentration_pb.MOLAR:
+        concentration_molar = concentration_pb.value
+    elif concentration_pb.units == concentration_pb.MILLIMOLAR:
+        concentration_molar = concentration_pb.value * 1e-3
+    elif concentration_pb.units == concentration_pb.MICROMOLAR:
+        concentration_molar = concentration_pb.value * 1e-6
+    # Assign moles amount and return.
+    moles = volume_liter * concentration_molar
+    if moles < 1e-6:
+        solute.moles.CopyFrom(reaction_pb2.Moles(units='NANOMOLES',
```

Comment:
I'd suggest making a copy of `solute` here before editing in case that solute is used elsewhere.

Reply:
Or possibly checking that solute doesn't already have a `moles` value set?

Reply:
What do you want the desired behavior to be here? I thought that an in-place modification of solute would actually be preferable.

Likewise, the intended behavior was to overwrite any previous value of `moles`. I could see one use case as something like...

```
for conc_string, yield_value in my_big_data_list:
  reaction = reaction_pb2.Reaction()
  reaction.CopyFrom(template_reaction)
  message_helpers.build_solution(
    reaction.inputs['solution A'].components[0],
    reaction.inputs['solution A'].components[1],
    conc_string
  )
  reaction.outcomes[0].products[0].yield.value = yield_value
  my_reactions.append(reaction)
```

For the case where we do not do any in-place modifications, that example would look like

```
for conc_string, yield_value in my_big_data_list:
  reaction = reaction_pb2.Reaction()
  reaction.CopyFrom(template_reaction)
  reaction.inputs['solution A'].components.MergeFrom(
    message_helpers.build_solution(
      reaction.inputs['solution A'].components[0],
      reaction.inputs['solution A'].components[1],
      conc_string
    )
  )
  reaction.outcomes[0].products[0].yield.value = yield_value
  my_reactions.append(reaction)
```

Thoughts?

Reply:
I see; I guess I was expecting that most people would be using `Compound` messages directly rather than having copied them into `reaction.inputs` first. I'm okay with keeping it as-is as long as we add some checks that the `moles` field hasn't already been set to avoid accidental overwrites. Does that sound good?

This ambiguity is another reason I might want to resolve #108 before merging this PR.

Reply:
Adding a keyword argument `overwrite` that defaults to `False` would be a good compromise here I think. In that case, an error would be raised if any of the `amount` fields of a `solute` are defined. Let's continue the broader discussion of solutions in #108

Reply:
SGTM